Insert-Into-ProjectList.pl

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/perl
+# This version is designed to run in the server2/analysis subdirectory
+# This file inserts projects into the projectlist database on the sql server
+
+
+use DBI;
+
+$input = "\n     Usage\:  Insert-Into-ProjectList.pl [projectNumber]\n\tMake sure the server config.xml is in the same directory as this Script.\n";
+$projectNumber = @ARGV[0] or die "$input";
+
+$projectXML = $projectNumber . ".xml";
+
+$projectNumber = substr $projectNumber, 4;
+
+$databaseServer = "'banana'";
+$server = "'Folding1'";
+
+$home_dir = "/home/server/server2";
+chomp($home_dir);
+$config_xml = "$home_dir/config.xml";
+
+# Opening the server xml inorder to find the full path to the projectXML
+open(INFILE, "$config_xml") or die "Can't open the file $config_xml\n";
+$projectFinder = 0;
+while(<INFILE>)
+{
+    @line = split;
+    if(index($line[1], $projectXML) != -1)
+    {
+        $projectFinder = 1;
+        $projectXML = substr $line[1], 6, -3;
+        print("Project's Full Path Found: " . $projectXML . "\n");
+        last; # Using "last" to exit out of the while loop
+    }
+}
+if($projectFinder == 0)
+{
+    print("Sory Project XML: " . $projectXML . " is not found \n");
+    die;
+}
+close(INFILE);
+
+# If full path is found, open the ProjectXML to set the variables
+open(INFILE, "$home_dir/$projectXML") or die "Can't open the file $projectXML\n";
+$projType_Finder = 0;
+while(<INFILE>)
+{
+    @line = split;
```

Comment:
Why not split on ``"``? Each value in the project.xml is written as \<attribute\> v=``"``Value``"``\</>

So the optimization would look something like:

```
while (my $line = <INFILE>)  {

    # split the current line on "
    my @values = split(/"/, $line); // This might need to be backslash escaped...
    if(index($values[], "title") != -1)
    {
          $description = "'". $line[1] . "'"'
     }
}
```
Example proj.xml
\<title v=``"``BChE with DP2``"``/>

Reply:
Above could be applied everywhere else (if it indeed works)...

Reply:
So if we split on " ", does that mean we will have 3 elements in the @values array? If that is the case, then I can implement that method instead.

Reply:
Yes. In the example I gave it would look like:
```
@values = ["<title v=", "BChE with DP2", "</>"]
```
Where ``$values[1] -> BChE with DP``

Reply:
Nice, that could've saved me so much lol. I'll implement that.

Reply:
Hahaha that's why I recommended it! Plus it's more general, which could prevent bugz in the future.